app/src/main/java/com/swent/mapin/model/chat/ConversationRepositoryFirestore.kt

```
@@ -0,0 +1,86 @@
+) : ConversationRepository {
+  /** Generates a new unique identifier for a conversation. */
+  override fun getNewUid(): String {
+    return db.collection("conversations").document().id
+  }
+  /**
+   * Observes conversations for the current user with live updates.
+   *
+   * @return A Flow emitting lists of [Conversation] objects.
+   */
+  override fun observeConversationsForCurrentUser(): Flow<List<Conversation>> = callbackFlow {
+    val uid =
+        auth.currentUser?.uid
+            ?: run {
+              close()
+              return@callbackFlow
+            }
+
+    val listener =
+        db.collection("conversations")
+            .whereArrayContains("participantIds", uid)
+            .orderBy("lastMessageTimestamp", Query.Direction.DESCENDING)
+            .addSnapshotListener { snapshot, error ->
+              if (error != null) {
+                close(error)
+                return@addSnapshotListener
+              }
+
+              val conversations =
+                  snapshot?.documents?.mapNotNull { doc ->
+                    val conversation = doc.toObject(Conversation::class.java)
+                    // Fallback ordering: if lastMessageTimestamp is null,
+                    // treat as very old (so it goes to the end)
+                    conversation
+                  } ?: emptyList()
+
+              trySend(conversations)
+            }
+
+    awaitClose { listener.remove() }
+  }
+  /**
+   * Adds a new conversation to Firestore.
+   *
+   * @param conversation The [Conversation] object to add.
+   */
+  override suspend fun addConversation(conversation: Conversation) {
+    val conversationRef = db.collection("conversations").document(conversation.id)
+
+    // Ensure the conversation includes the creator’s UID if not already added
+    val currentUid = auth.currentUser?.uid
+    val updatedParticipantIds =
+        if (currentUid != null && !conversation.participantIds.contains(currentUid)) {
```

Major:
Failing silently on null user hides authentication errors. Prefer explicit error reporting or early return with logging.
Silent exits make debugging Firestore sync issues harder.